lib/github.js

```diff
@@ -10,51 +10,32 @@ const GitHubApi = require('github')
  * @see {@link https://github.com/octokit/node-github}
  */
 
-// Default callback should just return the response passed to it.
 const defaultCallback = response => response
-
-class EnhancedGitHubClient extends GitHubApi {
-  constructor (options) {
-    super(options)
-    this.limiter = new Bottleneck(1, 1000)
-    this.logger = options.logger
+async function paginate (octokit, responsePromise, callback = defaultCallback) {
+  let collection = []
+  let getNextPage = true
+  let done = () => {
+    getNextPage = false
   }
+  let response = await responsePromise
+  collection = collection.concat(await callback(response, done))
+ // eslint-disable-next-line no-unmodified-loop-condition
+  while (getNextPage && octokit.hasNextPage(response)) {
+    response = await octokit.getNextPage(response)
+    collection = collection.concat(await callback(response, done))
+  }
+  return collection
+}
 
-  handler (params, block, callback) {
-    // Only allow one request at a time with a 1s delay
-    // https://github.com/octokit/node-github/issues/526
-    this.limiter.submit(super.handler.bind(this), params, block, (err, res) => {
-      let msg = `GitHub request: ${block.method} ${block.url}`
-      if (res) {
-        msg += ` - ${res.meta.status}`
-      } else if (err) {
-        msg += ` - ${err.code} ${err.status}`
-      }
-      this.logger.debug({params}, msg)
-
-      if (res) {
-        this.logger.trace(res, 'GitHub response:')
-      }
```

Comment:
FYI I removed the logging. I could add the logging back for when request succeed with `octokit.hook.after('request')` but there currently is no way to hook into request errors ... which might not be a bad idea to add 🤔

Reply:
```js
octokit.hook.after('request.error', (error, options) => {})
```

or

```js
octokit.hook.error('request', (error, options) => {})
```

Reply:
I like the first one, but I'm biased from Probot event handlers.

I do think the second one is more friendly. 🤔

Reply:
started a WIP pull request at https://github.com/gr2m/before-after-hook/pull/19. 
After giving it some thought, I like `octokit.hook.error` better, too